src/main/java/net/fabricmc/mappingio/format/proguard/ProGuardFileWriter.java

```diff
@@ -38,6 +38,7 @@ public final class ProGuardFileWriter implements MappingWriter {
 	private String srcName;
 	private String srcDesc;
 	private String dstName;
+	private List<String> argTypes = new ArrayList<>();
```

Comment:
Question: Why is this reused? Can extractArgumentTypes not just return a new list?

Reply:
Prevents unnecessary allocations, not sure how much of an impact it would actually have. Should I change it?

Reply:
I would, no need for premature optimisation.

Reply:
Done